CMakeLists.txt

```diff
@@ -11,7 +11,11 @@ find_package(YARP 2.3.64.13 REQUIRED)
 # Finding dependencies
 find_package(Boost REQUIRED serialization system)
 find_package(Protobuf REQUIRED)
-find_package(Gazebo   REQUIRED)
+find_package(Gazebo REQUIRED)
+if (Gazebo_VERSION_MAJOR LESS 5.0)
```

Comment:
It should be sufficient to just have `find_package(Gazebo 5.0 REQUIRED)`, see https://cmake.org/cmake/help/v3.0/command/find_package.html .